src/strategies/passport/passport-local/local-password-strategy-factory-provider.ts

```diff
@@ -29,79 +27,84 @@ export class LocalPasswordStrategyFactoryProvider
     return (options, verifier) =>
       this.getLocalStrategyVerifier(options, verifier);
   }
-
+  getLocalStrategyWithRequest(verifyFn: VerifyFunction.LocalPasswordFn) {
+    return async (
+      req: Request,
+      username: string,
+      password: string,
+      cb: (err: Error | null, user?: IAuthUser | false) => void,
+    ) => {
+      try {
+        const user = await verifyFn(username, password, req);
+        if (!user) {
+          throw new HttpErrors.Unauthorized(AuthErrorKeys.InvalidCredentials);
+        }
+        cb(null, user);
+      } catch (err) {
+        cb(err);
+      }
+    };
+  }
+  getLocalStrategyWithoutRequest(verifyFn: VerifyFunction.LocalPasswordFn) {
+    return async (
+      username: string,
+      password: string,
+      cb: (err: Error | null, user?: IAuthUser | false) => void,
+    ) => {
+      try {
+        const user = await verifyFn(username, password);
+        if (!user) {
+          throw new HttpErrors.Unauthorized(AuthErrorKeys.InvalidCredentials);
+        }
+        cb(null, user);
+      } catch (err) {
+        cb(err);
+      }
+    };
+  }
+  getLocalStrategyVerifierUndefinedRequest(
```

Comment:
name

Reply:
would 'getDefaultLocalStrategy' be work?

Reply:
yes